addons/godot-next/2d/vector_display_2d.gd

```diff
@@ -0,0 +1,49 @@
+tool
+class_name VectorDisplay2D
+extends Node
+# Displays Vector2 members in the editor via Position2D nodes.
+
+export(String) var variable_name
+export(bool) var relative = true
+
+var _storage: Node2D
+
+
+func _process(delta):
+	if not variable_name:
+		printerr("VectorDisplay2D: Please provide a variable name.")
+		return
+	
+	if not _storage:
+		_storage = Node2D.new()
+		get_tree().get_edited_scene_root().add_child(_storage)
+		return
+	
+	for child in _storage.get_children():
+		child.queue_free()
+	
+	var parent = get_parent()
+	if relative:
+		_storage.transform.origin = parent.global_transform.origin
+	else:
+		_storage.transform.origin = Vector2.ZERO
+	
+	var variable = parent.get(variable_name)
+	if variable == null:
+		printerr("VectorDisplay2D: Variable '" + variable_name + "' not found or invalid on parent node '" + get_parent().get_name() + "'.")
+	elif variable is Vector2:
+		_add_position_child(variable)
+	elif variable is PoolVector2Array:
+		for item in variable:
+			_add_position_child(item)
+	elif variable is Array:
+		for item in variable:
+			if item is Vector2:
+				_add_position_child(item)
+
+
+func _add_position_child(vector):
+	var node = Position2D.new()
+	node.transform.origin = vector
+	_storage.add_child(node)
+	node.set_owner(get_tree().get_edited_scene_root())
```

Comment:
Do we need to watch out for / account for situations where a scene might save that these generated nodes exist, and then close/re-open the scene only to have the nodes re-instantiated from scene load *and also* duplicated by this node's regular procedural generation operations?

Reply:
Surprisingly, no. These nodes aren't actually ever saved, they only appear at runtime. This behavior seems strange based on looking at the code, but... that's how it works.

Reply:
I think they *do* get added and saved (haven't tested it yet), because the logic with node ownership would imply that, but the code also goes through and deletes `_storage`'s children, which the generated nodes are, so I'm guessing it just ends up deleting and re-generating new ones *after* the scene loads them. Inconsequential though I think. It works<sup>TM</sup>.